ddtrace/contrib/anthropic/patch.py

```diff
@@ -0,0 +1,158 @@
+import json
+import os
+import sys
+from typing import Any
+
+import anthropic
+
+from ddtrace import config
+from ddtrace.contrib.trace_utils import unwrap
+from ddtrace.contrib.trace_utils import with_traced_module
+from ddtrace.contrib.trace_utils import wrap
+from ddtrace.internal.logger import get_logger
+from ddtrace.internal.utils import get_argument_value
+from ddtrace.llmobs._integrations import AnthropicIntegration
+from ddtrace.pin import Pin
+
+from .utils import _get_attr
+from .utils import record_usage
+
+
+log = get_logger(__name__)
+
+
+def get_version():
+    # type: () -> str
+    return getattr(anthropic, "__version__", "")
+
+
+config._add(
+    "anthropic",
+    {
+        "span_prompt_completion_sample_rate": float(os.getenv("DD_ANTHROPIC_SPAN_PROMPT_COMPLETION_SAMPLE_RATE", 1.0)),
+        "span_char_limit": int(os.getenv("DD_ANTHROPIC_SPAN_CHAR_LIMIT", 128)),
+    },
+)
+
+
+def _extract_api_key(instance: Any) -> str:
+    """
+    Extract and format LLM-provider API key from instance.
+    """
+    client = getattr(instance, "_client", "")
+    if client:
+        return getattr(client, "api_key", None)
+    return None
+
+
+@with_traced_module
+def traced_chat_model_generate(anthropic, pin, func, instance, args, kwargs):
+    chat_messages = get_argument_value(args, kwargs, 0, "messages")
+    integration = anthropic._datadog_integration
+
+    operation_name = func.__name__
+
+    span = integration.trace(
+        pin,
+        "%s.%s.%s" % (instance.__module__, instance.__class__.__name__, operation_name),
```

Comment:
Do you think the complete path (module, class, operation) may be excessive for naming spans? I'm happy to go with just class name + operation name, but up to you.

Reply:
Makes sense to me, it is a bit excessive rn